### PR TITLE
Get interior vehicle data implementation

### PR DESCRIPTION
**Pull request by LuxoftAKutsan**
Add GetInteriorVehicleData RPC implementation

### CODE REVIEW

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "can_cooperation/commands/get_interior_vehicle_data_request.h"
+#include "can_cooperation/validators/get_interior_vehicle_data_request_validator.h"
```
Comment (AKalinich-Luxoft) [inline src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:34]: @LuxoftAKutsan please remove these headers and all related files with validators as they are no longer used for request/response validation.

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+                 "conditional mandatory parameter "
+                     << kIsSubscribed << " missed in hmi response");
+    return;
+  }
+
+  if (!subscribe && hmi_response.isMember(kIsSubscribed)) {
+    LOG4CXX_WARN(logger_,
+                 "Parameter "
+                     << kIsSubscribed
+                     << " ignoring because it was not requested in request");
```
Comment (AKalinich-Luxoft) [inline src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:106]: @LuxoftAKutsan "ignored"

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+void GetInteriorVehicleDataRequest::OnEvent(
+    const can_event_engine::Event<application_manager::MessagePtr, std::string>&
+        event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  DCHECK_OR_RETURN_VOID(
+      (functional_modules::hmi_api::get_interior_vehicle_data == event.id()));
+  std::string info;
+
+  application_manager::Message& hmi_response = *(event.event_message());
+  bool validate_result = service()->ValidateMessageBySchema(hmi_response);
```
Comment (AByzhynar) [inline src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:73]: @LuxoftAKutsan const?

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        request_params[kModuleDescription]);
+  } else {
+    extension->UnsubscribeFromInteriorVehicleData(
+        request_params[kModuleDescription]);
+  }
+}
+
+std::string GetInteriorVehicleDataRequest::ModuleType(
+    const Json::Value& message) {
+  return message.get(message_params::kModuleDescription,
```
Comment (AKalinich-Luxoft) [inline src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:124]: @LuxoftAKutsan is it possible to use `operator[]` instead of get?

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+  SendRequest(functional_modules::hmi_api::get_interior_vehicle_data,
+              MessageHelper::StringToValue(message_->json_message()),
+              true);
+}
+
+void GetInteriorVehicleDataRequest::OnEvent(
+    const can_event_engine::Event<application_manager::MessagePtr, std::string>&
+        event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  DCHECK_OR_RETURN_VOID(
```
Comment (AByzhynar) [inline src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:68]: @LuxoftAKutsan I owuld rather use 
```
if (functional_modules::hmi_api::get_interior_vehicle_data == event.id()) {
...
} else {
    LOG4CXX_ERROR(logger_, "Received unknown event: " << event.id());
  }

```

File: src/components/interfaces/MOBILE_API.xml [hunk 6]
```diff
@@ -5755,7 +5756,7 @@
    </param>
 </function>
 
-  <function name="OnSeekMediaClockTimer" functionID="OnSeekMediaClockTimerID" messagetype="notification">
+<function name="OnSeekMediaClockTimer" functionID="OnSeekMediaClockTimerID" messagetype="notification">
```
Comment (AKalinich-Luxoft) [inline src/components/interfaces/MOBILE_API.xml:5759]: @LuxoftAKutsan looks not related to this pull request

**Review by AKalinich-Luxoft — CHANGES_REQUESTED** [review]

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc (hunk 2 above)
Comment (AByzhynar) [inline src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:106]: @LuxoftAKutsan `ignoring because it was not requested in request` -> `is ignored due to absence in request"`

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+  if (!subscribe && hmi_response.isMember(kIsSubscribed)) {
+    LOG4CXX_WARN(logger_,
+                 "Parameter "
+                     << kIsSubscribed
+                     << " ignoring because it was not requested in request");
+    return;
+  }
+
+  bool is_subscribed = response_params_[kIsSubscribed].asBool();
```
Comment (AByzhynar) [inline src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:110]: @LuxoftAKutsan const?

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: src/components/interfaces/HMI_API.xml [hunk 8]
```diff
@@ -4725,6 +4725,28 @@
    <param name="interiorVehicleDataCapabilities" type="Common.RemoteControlCapabilities">
    </param>
  </function>
+
+<function name="GetInteriorVehicleData" functionID="GetInteriorVehicleDataID" messagetype="request">
+    <param name="moduleDescription" type="Common.ModuleDescription">
+      <description>The zone and module data to retrieve from the vehicle for that zone</description>
```
Comment (AByzhynar) [inline src/components/interfaces/HMI_API.xml:4731]: @LuxoftAKutsan MAy be remove `zones` from description?

**Review by AByzhynar — CHANGES_REQUESTED** [review]

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc (hunk 4 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:124]: @AKalinich-Luxoft operator[]  is not const

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc (hunk 5 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:68]: @AByzhynar I strongly think that in that case DCHECK in better.

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc (hunk 3 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:73]: @AByzhynar 16f9265

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc (hunk 7 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:110]: @AByzhynar 16f9265

File: src/components/interfaces/HMI_API.xml (hunk 8 above)
Comment (LuxoftAKutsan) [reply src/components/interfaces/HMI_API.xml:4731]: @AByzhynar 16f9265

File: src/components/interfaces/MOBILE_API.xml (hunk 6 above)
Comment (LuxoftAKutsan) [reply src/components/interfaces/MOBILE_API.xml:5759]: @AByzhynar 16f9265

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc (hunk 2 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:106]: @AByzhynar 16f9265

File: src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc (hunk 1 above)
Comment (LuxoftAKutsan) [reply src/components/can_cooperation/src/commands/get_interior_vehicle_data_request.cc:34]: @AByzhynar 16f9265

**Review by AKalinich-Luxoft — APPROVED** [review]

**Review by AByzhynar — APPROVED** [review]
